### DISCUSSION TITLE
Supported Languages

### DISCUSSION THREAD

**Discussion by diannegabriel** [markdoc/markdoc#456] [Q&A] [answered]
Hi there,

Which languages do Markdoc support when it comes to syntactical highlighting for code blocks? I'm trying to write documentation and one of the code blocks I'm writing is for GraphQL.

What I wrote (ignore backslash):
```
\```graphql
type Person {
  name: String!
  age: Int!
}

type Post {
  title: String!
}

<!-- The '!' after the type means its non-nullable -->
```\
```

What I have on my VSCode:
![image](https://github.com/markdoc/markdoc/assets/84348468/d0fa9aa5-2263-469c-8e3c-10228bb347d8)

Result:
<img width="603" alt="image" src="https://github.com/markdoc/markdoc/assets/84348468/4bd4c8d0-2dea-4aec-abe4-1b5662119eef">

The code block isn't being formatted properly. I went to look at Prism and noticed [GraphQL is a supported language](https://prismjs.com/index.html#supported-languages). 

Other languages I've tested are JS, shell, and CSS, all are working properly. Latest version of Prism is installed.

Any input is appreciated. Thank you!

**Comment by mfix-stripe** [comment, marked as answer]
Hey @diannegabriel, supported languages is a consideration separate from Markdoc. With Markdoc, you can use any syntax highlighting framework, so depending on which one you choose, different languages will be supported. 

In [our starter](https://github.com/markdoc/markdoc-starter), we use Prism.js, which supports these languages: https://prismjs.com/#supported-languages

Hope that helps!
